asf_public_discourse_home_decarbonisation/pipeline/bert_topic_analysis/evaluate_bertopic_results.py

```diff
@@ -0,0 +1,367 @@
+        argparse.Namespace: parsed arguments
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--n_runs", type=int, help="number of runs", default=10)
+    parser.add_argument(
+        "--path_to_config_file",
+        type=str,
+        help="path to params config file",
+        default="asf_public_discourse_home_decarbonisation.pipeline.bert_topic_analysis.bert_params_config",
+    )
+    parser.add_argument(
+        "--path_to_data_file",
+        type=str,
+        help="Path to data file, if not standard forum data.",
+        default=None,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def get_configuration_params_file(path_to_config_file: str) -> tuple:
+    """
+    Loads file with configuration parameters including:
+    - data source parameters
+    - model and additional parameters
+    And returns a tuple of two dictionaries with these configurations.
+
+    Args:
+        path_to_config_file (str): Path to configuration file in the repo structure
+        e.g. asf_public_discourse_home_decarbonisation.pipeline.bert_topic_analysis.bert_params_config
+
+    Returns:
+        tuple: data source parameters and model/additional parameters
+    """
+    config_module = importlib.import_module(path_to_config_file)
+
+    data_source_parms = config_module.data_source_parms
+    model_and_additional_params = config_module.model_and_additional_params
+
+    return data_source_parms, model_and_additional_params
+
+
+def create_boxplots_with_results(output_configs: dict):
+    """
+    Creates and saves a figure with boxplots showing the distribution of results for:
+    - Number of topics
+    - Percentage of outliers
+    - Average probability of beloging to a non-outlier topic
+
+    Args:
+        output_configs (dict): dictionary with results information
+    """
+
+    # Extracting information from output_configs dictionary
+    source_name = output_configs["data_source"]
+    category = output_configs["category"]
+    n_docs = output_configs["n_docs"]
+    n_runs = output_configs["n_runs"]
+    keywords = output_configs["keywords"]
+    number_of_outliers_df = output_configs["outliers_df"]
+    number_of_topics_df = output_configs["topics_df"]
+    avg_probablity_df = output_configs["probabilities_df"]
+    outliers_vs_docs_dict = output_configs["outliers_vs_docs_dict"]
+
+    # Plot horizontal boxplots for each dataframe: number_of_topics_df, number_of_outliers_df and avg_probablity_df
+    fig, axes = plt.subplots(3, 1, figsize=(10, number_of_topics_df.shape[1] * 3))
+
+    number_of_topics_df.boxplot(ax=axes[0], grid=False, vert=False)
+    axes[0].set_xlabel("Number of topics")
+
+    percentage_of_outliers_df = number_of_outliers_df / n_docs * 100
+    percentage_of_outliers_df.boxplot(ax=axes[1], grid=False, vert=False)
+    axes[1].set_xlabel("Percentage of outliers")
+    # Making the x-axis always between 0 and 50 (just as visual aid)
+    if percentage_of_outliers_df.max().max() <= 50:
```

Comment:
If percentage of outliers is greater than 50%, does it plot with 0-100% axis I assume?

Reply:
no, i think it plots from 0 to max found (or max found + a bit more)